core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -543,7 +547,8 @@ class KafkaServer(
           clusterId = clusterId,
           time = time,
           tokenManager = tokenManager,
-          apiVersionManager = apiVersionManager)
+          apiVersionManager = apiVersionManager,
+          clientMetricsManager = clientMetricsManager)
```

Comment:
Since client metric is not supported in ZK mode, should we make clientMetricsManager an Option and pass in None here?

Reply:
I agree, it makes sense to not use KafkaServer as that's specifically in ZK mode. I have made the param optional and pushed the change.